### DISCUSSION TITLE
Could not load System.Memory, Version=4.0.1.1  from assembly: Microsoft.Data.SqlClient, Version=5.0.0.0

### DISCUSSION THREAD

**Discussion by jssmotta** [dotnet/SqlClient#2856] [General]
Message: 
Could not load file or assembly 'System.Memory, Version=4.0.1.1 ... Calling from assembly: Microsoft.Data.SqlClient, Version=5.0.0.0

Environment:
.NET 4.8.1 WebForms App; (It happens in 4.7.x too);
SQL Server 2022;
VS 2022 (and Preview too);

The throw error when:

` var oCnn = new SqlConnection("Server=localhost;Database=AdventureWorks;Integrated Security=True;");
oCnn.Open();`

What was done to try to fix:

GitHub Copilot does not answer correctly to fix it.

Redirects do not work.

Reinstalling System.Memory does not work.

Installing SNI does not work.

**Comment by JRahnama** [comment]
what version of M.D.SqlClient are you using? 5.2.2 or 5.1 or trying the v6.0-preview1?

**Reply by jssmotta** [reply]
The last stable!

**Reply by jssmotta** [reply]
I mean 5.2.2.
